internal/content_coding.go

```diff
@@ -9,10 +9,25 @@ import (
 
 	"github.com/klauspost/compress/gzip"
 	"github.com/klauspost/compress/zlib"
+	"github.com/klauspost/compress/zstd"
 	"github.com/klauspost/pgzip"
 )
 
-const DefaultMaxDecompressionSize = 500 * 1024 * 1024 //500MB
+const DefaultMaxDecompressionSize int64 = 500 * 1024 * 1024 //500MB
+
+// DecodingOption provide methods to change the decoding from the standard
+// configuration.
+type DecodingOption func(*decoderConfig)
+
+type decoderConfig struct {
+	maxDecompressionSize int64
+}
+
+func DecoderMaxDecompressionSize(maxDecompressionSize int64) DecodingOption {
```

Comment:
Maybe name it `WithSizeLimit` or something?

Reply:
I agree with that I named it like this because of the Encoder part but looking at the standards it seems `With` keyword is used everywhere.